src/typing_manager.ts

```diff
@@ -52,6 +52,21 @@ export class TypingManager implements Disposable {
         this.disposables.push(commands.registerCommand("vscode-neovim.send", this.onSendCommand));
         this.disposables.push(commands.registerCommand("vscode-neovim.send-blocking", this.onSendBlockingCommand));
         this.disposables.push(commands.registerCommand("vscode-neovim.escape", this.onEscapeKeyCommand));
+        this.disposables.push(
+            commands.registerCommand("vscode-neovim.toggle", () => {
+                this.main.modeManager.neovimToggle = !this.main.modeManager.neovimToggle;
+                let batch: [string, ...unknown[]][];
+                if (!this.main.modeManager.neovimToggle) {
+                    this.logger.debug(`${LOG_PREFIX}: Disable Vscode-neovim`);
+                    batch = [["mode_change", ["insert"]]];
```

Comment:
I don't have time to try right now, but before I merge this I'll want to take a look to see if there is a better way to enter insert mode. This seems like a bit of a hack, ideally we just tell nvim to enter/leave insert mode directly using `startinsert` `stopinsert`.

Reply:
would be highly appreciated.